src/drivers/uavcan_v1/templates/_composite_type.j2

```diff
@@ -0,0 +1,115 @@
+{{ composite_type | definition_begin }}
+{
+{%- if T.fixed_port_id is not None %}
+	static constexpr CanardPortID PORT_ID = {{T.fixed_port_id}};
+{%- endif %}
+
+{%- set total_size = namespace(value=0) -%}
+{%- for field in composite_type.fields -%}
+    {% set total_size.value = total_size.value + field.data_type.bit_length_set|max %}
+{%- endfor %}
+	static constexpr size_t SIZE = {{total_size.value}};
+
+	static constexpr auto getDataTypeFullName() { return "{{T}}"; }
+
+{% for constant in composite_type.constants %}
+	static constexpr {{ constant.data_type | declaration }} {{ constant.name | id }} = {{ constant.value.native_value.numerator }} / {{ constant.value.native_value.denominator }};
```

Comment:
An esoteric gotcha here: you'll be performing floating-point operations in the compiler here which means the precision can be different based on the machine and/or toolchain that compiled the code. This is probably fine for lower design-assurance level code but once you get to high-energy, commercial-aircraft levels of safety you'll need to choose a strategy for ensuring deterministic calculation of constant values between sub-systems utilizing these values. My plan for Nunavut was to either allow runtime initialization of these "constants" (which only solves the problem if all the FPUs on the bus are the same) or to provide some sort of software emulation that provide pre-defined, consistent results. Unfortunately I haven't given much more thought to it than that.